docs/kns_apps_tutorial.md

```diff
@@ -0,0 +1,347 @@
+        "Hello": {
+```
+
+* `commands` is a list, so you can add any app entry points as needed. This can be useful to create an all-in-one app image for example.
+* `hello` is the name of the current entry point for the app.
+
+```json
+        "Hello": {
+            "path": "/usr/bin/true",
+            "name": "Start nested http server",
+            "description": "Start a nested http server for testing and to say hello world.",
+            "parameters": {},
+```
+
+For KNS jobs, `path` and `parameters` are not used, but need to be present for the app to be usable by Jarvice upstream component.
+`path` should be set to `/usr/bin/true` and `parameters` left as an empty dict.
+
+`name` and `description` will display as before some details about this entry point, for users to be able to choose between multiple if exist.
+
+```json
+            "nestedkubernetes": {
+                "kubernetes_version": "",
+                "enable_gotty_shell": "true",
+                "ingress_type": "subdomain",
+                "targets": [
+                    {
+                        "service_name": "web",
+                        "service_port": 8080,
+                        "service_namespace": "default"
+                    }
+                ],
+                "templates_settings": {
+                    "global_retry": 20,
+                    "global_retry_sleep": 15
+                },
+                "templates": [
+                    {
+                        "name": "google-sample",
+                        "type": "kubectl",
+                        "path": "google-sample.yml"
+                    }
+                ]
+            }
+```
+
+`nestedkubernetes` is the new key that allows to define inside it all needed elements for a Kubernetes based app.
+Lets see each parts of this new dict.
+
+### 2.1. General parameters
+
+```json
+            "nestedkubernetes": {
+                "kubernetes_version": "",
+```
+
+* `kubernetes_version` allows to specify the Kubernetes version to be used for the K3S nested cluster. Format is `v1.X.Y` with `X` the major and `Y` the minor versions. For example: `v1.28.11`. See the compatibility matrix here: https://vcluster.com/docs/v0.19/deploying-vclusters/compat-matrix
+
+### 2.2. Ingress parameters
+
+Ingress parameters allow to connect app to the external world, by giving final user the URL to reach the app portal/entry point/API/etc.
+
+```json
+                "ingress_type": "subdomain",
+```
+
+`ingress_type` specify the type of ingress to be use. By default it is `subdomain`, which means that for each job, the ingress URL will be of format kns-job-XXX.mycluster.example.
+On some clusters, a subdomain based ingress cannot be used, and so instead of providing a subdomain based ingress, the url will be of format mycluster.example/XXX. To do so, set `ingress_type` to `prefix`. Note that not all KNS applications are compatible with this workaround (Kubeflow for example will not work).
```

Comment:
```suggestion
On some clusters, a subdomain based ingress cannot be used, and so instead of providing a subdomain based ingress, the URL will be of format mycluster.example/XXX. To do so, set `ingress_type` to `prefix`. Note that not all KNS applications are compatible with this workaround (Kubeflow for example will not work).
```